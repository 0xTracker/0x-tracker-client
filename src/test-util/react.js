import { createMemoryHistory } from 'history';
import { render } from '@testing-library/react';
import { Router } from 'react-router-dom';
import React from 'react';

function renderWithRouter(
  element,
  {
    route = '/',
    history = createMemoryHistory({ initialEntries: [route] }),
  } = {},
) {
  return {
    ...render(<Router history={history}>{element}</Router>),
    history,
  };
}

export { renderWithRouter };
