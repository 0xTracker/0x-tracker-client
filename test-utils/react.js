import { createMemoryHistory } from 'history';
import { createStore } from 'redux';
import { Provider } from 'react-redux';
import { render } from 'react-testing-library';
import { Router } from 'react-router-dom';
import React from 'react';

import reducer from '../src/redux/reducer';

function renderWithRouter(
  element,
  {
    route = '/',
    history = createMemoryHistory({ initialEntries: [route] }),
  } = {},
) {
  return {
    ...render(<Router history={history}>{element}</Router>),
    history,
  };
}

const renderWithAppContext = (
  component,
  {
    route = '/',
    history = createMemoryHistory({ initialEntries: [route] }),
    initialState,
    store = createStore(reducer, initialState),
  } = {},
) => ({
  ...render(
    <Provider store={store}>
      <Router history={history}>{component}</Router>
    </Provider>,
  ),
  history,
});

export { renderWithAppContext, renderWithRouter };
