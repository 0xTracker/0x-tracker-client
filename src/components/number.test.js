import { render } from '@testing-library/react';
import React from 'react';

import Number from './number';

it('should not format 999', () => {
  const { asFragment } = render(<Number>{999}</Number>);

  expect(asFragment()).toMatchSnapshot();
});

it('should format 1000', () => {
  const { asFragment } = render(<Number>{1000}</Number>);

  expect(asFragment()).toMatchSnapshot();
});

it('should format 10000', () => {
  const { asFragment } = render(<Number>{10000}</Number>);

  expect(asFragment()).toMatchSnapshot();
});

it('should format 100000', () => {
  const { asFragment } = render(<Number>{100000}</Number>);

  expect(asFragment()).toMatchSnapshot();
});

it('should format 10000000', () => {
  const { asFragment } = render(<Number>{1000000}</Number>);

  expect(asFragment()).toMatchSnapshot();
});

it('should summarize 4530', () => {
  const { asFragment } = render(<Number summarize>{4530}</Number>);

  expect(asFragment()).toMatchSnapshot();
});

it('should summarize 152650', () => {
  const { asFragment } = render(<Number summarize>{152650}</Number>);

  expect(asFragment()).toMatchSnapshot();
});

it('should summarize 7846032', () => {
  const { asFragment } = render(<Number summarize>{7846032}</Number>);

  expect(asFragment()).toMatchSnapshot();
});
