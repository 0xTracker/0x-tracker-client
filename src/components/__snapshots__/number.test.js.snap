// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should format 1000 1`] = `
<DocumentFragment>
  1,000
</DocumentFragment>
`;

exports[`should format 10000 1`] = `
<DocumentFragment>
  10,000
</DocumentFragment>
`;

exports[`should format 100000 1`] = `
<DocumentFragment>
  100,000
</DocumentFragment>
`;

exports[`should format 10000000 1`] = `
<DocumentFragment>
  1,000,000
</DocumentFragment>
`;

exports[`should not format 999 1`] = `
<DocumentFragment>
  999
</DocumentFragment>
`;

exports[`should summarize 4530 1`] = `
<DocumentFragment>
  <span
    title="4,530"
  >
    4.5k
  </span>
</DocumentFragment>
`;

exports[`should summarize 152650 1`] = `
<DocumentFragment>
  <span
    title="152,650"
  >
    152.7k
  </span>
</DocumentFragment>
`;

exports[`should summarize 7846032 1`] = `
<DocumentFragment>
  <span
    title="7,846,032"
  >
    7.8m
  </span>
</DocumentFragment>
`;
